File: Campaign_Samples/js/create_campaign.js
var request = require("request");
var accountSid = "<your Exotel Sid>";
var accountToken = "<your Exotel Token>";
var encoding = Buffer.from(accountSid + ':' + accountToken).toString('base64')
var options = { method: 'POST',
  url: 'https://api.exotel.com/v2/accounts/'+ accountSid + '/campaigns',
  headers: 
   { 
     Authorization: 'Basic ' + encoding,
     'Content-Type': 'application/json' },
  body: 
   { campaigns: 
      [ { caller_id: '0XXXXXXXXX1',
          url: 'http://my.exotel.in/start/189810',
          from: [ '+91XXXXXXXXX4', '+91XXXXXXXXX5' ],
          schedule: { send_at: '2018-11-08T14:20:00+05:30' },
          status_callback: 'http://<callback custom domain>/1gvta9f1',
          call_status_callback: 'http://<callback custom domain>/1gvta9f1' } ] },
  json: true };

request(options, function (error, response, body) {

  if (error) throw new Error(error);
  console.log(body.error_data)
  console.log(body)
});

File: Campaign_Samples/go/create_campaign.go
package main

import (
	b64 "encoding/base64"
	"fmt"
	"io/ioutil"
	"net/http"
	"strings"
)

func main() {

	// Please provide accountSid, authToken from your Exotel account
	accountSid := "<your Exotel Sid>"
	apiKey := "<yout API Key>"
	apiToken := "<your Exotel Token>"

	// Encoding the accountSid and authToken, used in Authorization header
	encoding := b64.StdEncoding.EncodeToString([]byte(apiKey + ":" + apiToken))

	url := "https://api.exotel.com/v2/accounts/" + accountSid + "/campaigns"

	payload := strings.NewReader(`{ 
		"campaigns": [{ 
			"caller_id": "0XXXXXXXXX1", 
			"url": "http://my.exotel.com/exoml/start_voice/208287", 
			"from": [ "+91XXXXXXXXX4", "+91XXXXXXXXX5" ], 
			"status_callback": "http://<callback custom domain>/1gvta9f1", 
			"call_status_callback": "http://<callback custom domain>/1gvta9f1"
			}]
		}`)

	req, _ := http.NewRequest("POST", url, payload)

	req.Header.Add("Content-Type", "application/json")
	req.Header.Add("Authorization", "Basic "+encoding)

	res, _ := http.DefaultClient.Do(req)

	defer res.Body.Close()
	body, _ := ioutil.ReadAll(res.Body)

	fmt.Println(res)
	fmt.Println(string(body))

}
